scss/components/button.scss

```diff
@@ -116,6 +116,19 @@ a.#{$block} {
       }
     }
   }
+  &--half {
+    --fd-forms-height-half: #{$fd-forms-height/2};
+    @include fd-var-size("height", $fd-forms-height/2, --fd-forms-height-half);
+    @include fd-var-size("max-height", $fd-forms-height/2, --fd-forms-height-half);
+    line-height: 1;
+
+    &.#{$block}--compact {
+      --fd-forms-height-half: #{$fd-forms-height--compact/2};
+      @include fd-var-size("height", $fd-forms-height--compact/2, --fd-forms-height-half);
+      @include fd-var-size("max-height", $fd-forms-height--compact/2, --fd-forms-height-half);
+      line-height: 1;
+    }
+  }
```

Comment:
Added to allow buttons to support the stacked up/down on the number "widget" with input group.